File: cpp_e9zMZ/cpp/10-tonghop/75.cpp
#include <iostream>
using namespace std;

int main() {
    string str = "hoclaptrinh";
    string key = "335";
    string result = "";
    int startIndex = 0;

    for (int i = 0; i < key.length(); i++) {
        int size = key[i] - '0';
        string word = str.substr(startIndex, size) + " ";
        word[0] = toupper(word[0]);
        result += word;
        startIndex += size;
    }

    cout << result << endl;
    return 0;
}

File: python_ToSXs/python/10-tonghop/75.py
str = "luutruonghailan"
key = "3633"

result = ""
startIndex = 0
endIndex = 0

for elm in key:
    currentNumber = int(elm)
    endIndex += currentNumber

    word = str[startIndex:endIndex]
    result += word.capitalize() + " "
    startIndex = endIndex

print(result.strip())
